astropy/wcs/utils.py

```diff
@@ -95,10 +100,48 @@ def _wcs_to_celestial_frame_builtin(wcs):
     return frame
 
 
+def _celestial_frame_to_wcs_builtin(frame, projection='TAN'):
+
+    # Import astropy.coordinates here to avoid circular imports
+    from ..coordinates import BaseRADecFrame, FK4, FK4NoETerms, FK5, ICRS, Galactic
+
+    # Create a 2-dimensional WCS
+    wcs = WCS(naxis=2)
+
+    if isinstance(frame, BaseRADecFrame):
+
+        xcoord = 'RA--'
+        ycoord = 'DEC-'
+        if isinstance(frame, ICRS):
+            wcs.wcs.radesys = 'ICRS'
+        elif isinstance(frame, FK4NoETerms):
+            wcs.wcs.radesys = 'FK4-NO-E'
+            wcs.wcs.equinox = frame.equinox.byear
+        elif isinstance(frame, FK4):
+            wcs.wcs.radesys = 'FK4'
+            wcs.wcs.equinox = frame.equinox.byear
+        elif isinstance(frame, FK5):
+            wcs.wcs.radesys = 'FK5'
+            wcs.wcs.equinox = frame.equinox.jyear
+        else:
+            return None
+    elif isinstance(frame, Galactic):
+        xcoord = 'GLON'
+        ycoord = 'GLAT'
+    else:
+        return None
+
+    wcs.wcs.ctype[0] = xcoord + '-' + projection
+    wcs.wcs.ctype[1] = ycoord + '-' + projection
+
```

Comment:
@astrofrog This does not set any projection attributes, like `lonpole` and `latpole`. For example, for `TAN` projection `lonpole` is always 180 deg. This can be fixed by adding a line here
```
wcs.wcs.set()
```
You can see the difference in computed values like this:
```
frame = ICRS()
mywcs = utils.celestial_frame_to_wcs(frame)
mywcs.all_pix2world(1,1,1)
Out[55]: [array(1.0), array(1.0)]
mywcs.wcs.lonpole
Out[56]: nan

mywcs = utils.celestial_frame_to_wcs(frame)
mywcs.wcs.set()
mywcs.wcs.lonpole
Out[59]: 180.0
mywcs.all_pix2world(1,1,1)
Out[60]: [array(0.9998984794143875), array(0.999746251856687)]


```